x-pack/plugins/grokdebugger/server/routes/api/grokdebugger/register_grok_simulate_route.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { GrokdebuggerRequest } from '../../../models/grokdebugger_request';
+import { GrokdebuggerResponse } from '../../../models/grokdebugger_response';
+
+const requestBodySchema = schema.object({
+  pattern: schema.string(),
+  rawEvent: schema.string(),
+  // We don't know these key / values up front as they depend on user input
+  customPatterns: schema.object({}, { unknowns: 'allow' }),
+});
+
+export function registerGrokSimulateRoute(framework) {
+  framework.registerRoute(
```

Comment:
I am not convinced we need to wrap both license state and the http service in a class. IRouter has quite a well designed API so my suggestion would be to create the router on the plugin and pass that in as a route dependency. Then we need to just wrap the route handler. For example see `x-pack/plugins/index_management/server/services/license.ts`.